core/src/main/scala/filodb.core/memstore/IndexBootstrapper.scala

```diff
@@ -56,21 +56,24 @@ class IndexBootstrapper(colStore: ColumnStore) {
   def bootstrapIndexDownsample(index: PartKeyLuceneIndex,
                                shardNum: Int,
                                ref: DatasetRef,
+                               checkpointMillis: Option[Long],
                                ttlMs: Long)
                               (assignPartId: PartKeyRecord => Int): Task[Long] = {
 
     // This is where we need to only get the delta from  PartKeyLuceneIndex and fetch part keys updated after
     // the timestamp in millis, based on the time, we need to invoke refreshWithDownsamplePartKeys giving
     // the last synced time till current hour, the start hour will be max(of the time retrieved from underlying
     // state, start - ttlMs)
-
     val recoverIndexLatency = Kamon.gauge("shard-recover-index-latency", MeasurementUnit.time.milliseconds)
       .withTag("dataset", ref.dataset)
       .withTag("shard", shardNum)
     val start = System.currentTimeMillis()
     //here we need to adjust ttlMs
     //because we might already have index available on disk
-    val checkpointTime = start - ttlMs
+    var checkpointTime = checkpointMillis.getOrElse(0L)
```

Comment:
Check previous comment on what bootstrap can not accept a start time to refresh from yet.  Unless dependency on partIds is removed.

Reply:
Added a comment. The code provides the capability to recover from a checkpoint when we add missing functionality. Essentially, here we are adding a skeleton from the  future implementation.